custom/lib/Passage.php

```diff
@@ -299,9 +300,21 @@ public function updateUser(
         trigger_error('Method ' . __METHOD__ . ' is deprecated', E_USER_DEPRECATED);
 
         try {
-            return $this->user->update($user_id, $update_user_request);
+            $args = $this->cast($update_user_request, UpdateUserArgs::class);
+            return $this->user->update($user_id, $args);
         } catch (PassageError $e) {
             throw $e->getPrevious();
         }
     }
+
+    private function cast(object $object, string $class): mixed
```

Comment:
this function serializes a class into a string, adjusts the class name, and unserializes it back into the new class. without this the functions will throw errors at runtime that it was expecting one class but was provided another.

Reply:
Just to check, a pattern to achieve something similar in Go sometimes is to turn a input struct to JSON & deserialize it to the target struct. Would it be possible to do something similar here? Like turn the input class into a JSON string or an associative array, to construct the target class?

Or is the pattern below the PHP equivalent of that?

Reply:
@bertrmz can you check in here to make sure this answered your question.

Reply:
👍 seems to all load as expected against the current test suite. We will need to update the suite to use these new classes soon, but sounds like you've had a chance to run through that in local testing.